src/main/scala/se/nimsa/sbx/dicom/DicomMetaDataDAO.scala

```diff
@@ -294,6 +294,40 @@ class DicomMetaDataDAO(val driver: JdbcProfile) {
   }
 
   def studies(implicit session: Session): List[Study] = studiesQuery.list
+  
+  def queryStudies(startIndex: Long, count: Long, orderBy: Option[String], orderAscending: Boolean, queryProperties: Seq[QueryProperty])(implicit session: Session): List[Study] = {
+    implicit val getResult = GetResult(r =>
+      Study(r.nextLong, r.nextLong, StudyInstanceUID(r.nextString), StudyDescription(r.nextString), StudyDate(r.nextString), StudyID(r.nextString), AccessionNumber(r.nextString), PatientAge(r.nextString)))
+    
+    var query = """select distinct("Studies"."id"),
+      "Studies"."patientId",
+      "Studies"."StudyInstanceUID",
+      "Studies"."StudyDescription",
+      "Studies"."StudyDate",
+      "Studies"."StudyID",
+      "Studies"."AccessionNumber",
+      "Studies"."PatientAge" from "Studies"
+      inner join "Patients" on "Patients"."id" = "Studies"."patientId"
+      inner join "Series" on "Series"."studyId" = "Studies"."id"
+      inner join "Equipments" on "Equipments"."id" = "Series"."equipmentId"
+      inner join "FrameOfReferences" on "FrameOfReferences"."id" = "Series"."frameOfReferenceId""""
+    
+    var wherePart = ""
+    
+    queryProperties.foreach(queryProperty => {
```

Comment:
OM du vill vara super scala-esque här så kan man säkert skriva det som 
queryProperties.map(queryProperty => s""""${queryProperty.propertyName}" ${queryProperty.operator.toString()} '${queryProperty.propertyValue}'""").mkString(" and ")

eller om du vill göra det mer lättläst så definiera först en lokal funktion
def queryPropertyToPart(queryProperty: QueryProperty) = 
  s""""${queryProperty.propertyName}" ${queryProperty.operator.toString()} '${queryProperty.propertyValue}'"""

och sedan 

queryProperties.map(queryPropertyToPart(_)).mkString(" and ")